### PR TITLE
[MRG] expose feature importance to c_api

### PR DESCRIPTION
**Pull request by wxchan**
not let python feature importance rely on dumped json model

@guolinke there are two feature importance types in python now, they are different return types (split - int, gain - float), what do you think it's a better way to deal with it, should I make them all return float?

### CODE REVIEW

**Comment by guolinke** [comment]
@wxchan I think returning float/double is okay.

**Comment by guolinke** [comment]
@wxchan 
can this fix the importance of single-leaf tree?

File: src/boosting/gbdt.cpp [hunk 1]
```diff
@@ -1130,30 +1145,35 @@ bool GBDT::LoadModelFromString(const std::string& model_str) {
...
 
-  std::vector<size_t> feature_importances(max_feature_idx_ + 1, 0);
-  for (int iter = 0; iter < num_used_model; ++iter) {
-    for (int split_idx = 0; split_idx < models_[iter]->num_leaves() - 1; ++split_idx) {
-      if (models_[iter]->split_gain(split_idx) > 0) {
-        ++feature_importances[models_[iter]->split_feature(split_idx)];
+  std::vector<double> feature_importances(max_feature_idx_ + 1, 0.0);
+  if (importance_type == 0) {
+    for (int iter = boost_from_average_ ? 1 : 0; iter < num_used_model; ++iter) {
+      for (int split_idx = 0; split_idx < models_[iter]->num_leaves() - 1; ++split_idx) {
```
Comment (wxchan) [inline src/boosting/gbdt.cpp]: @guolinke the single leaf tree will be skipped by here. I think the previous error happens on dump model. Can you find a stable reproduced example?

Comment (guolinke) [inline src/boosting/gbdt.cpp]: @wxchan I cannot find a stable reproduced example. But all the errors are the json decode errors.

Comment (StrikerRUS) [inline src/boosting/gbdt.cpp]: @wxchan Travis also sometimes randomly fails on `test_feature_importances_single_leaf`: [example 1](https://travis-ci.org/StrikerRUS/LightGBM/builds/267123947), [2](https://travis-ci.org/StrikerRUS/LightGBM/builds/266854430).

Comment (wxchan) [reply src/boosting/gbdt.cpp]: @StrikerRUS ok, didn't find out reason yet, it's good to know what char is there.

**Pull request merged**